pkg/build/builder/source.go

```diff
@@ -56,38 +65,55 @@ func fetchSource(dir string, build *api.Build, urlTimeout time.Duration, in io.R
 	return nil
 }
 
-// checkSourceURI performs a check on the URI associated with the build
-// to make sure that it is valid.  It also optionally tests the connection
-// to the source uri.
-func checkSourceURI(git git.Git, rawurl string, testConnection bool, timeout time.Duration) error {
-	if !git.ValidCloneSpec(rawurl) {
-		return fmt.Errorf("Invalid git source url: %s", rawurl)
-	}
-	if strings.HasPrefix(rawurl, "git@") || strings.HasPrefix(rawurl, "git://") {
-		return nil
-	}
-	srcURL, err := url.Parse(rawurl)
+// checkRemoteGit validates the specified Git URL. It returns GitNotFoundError
+// when the remote repository not found and GitAuthenticationError when the
+// remote repository failed to authenticate.
+// Since this is calling the 'git' binary, the proxy settings should be
+// available for this command.
+func checkRemoteGit(url string, timeout time.Duration) error {
+	glog.V(3).Infof("git ls-remote --heads %q", url)
+	cmd := exec.Command("git", "ls-remote", "--heads", url)
+	cmd.Env = []string{"GIT_ASKPASS=/bin/true"}
+
+	var (
+		out []byte
+		err error
+	)
+
+	finish := make(chan struct{}, 1)
+	go func() {
+		out, err = cmd.CombinedOutput()
```

Comment:
You're eating the error to invoke.  Make that explicit that you eat and ignore it